classes/phing/Phing.php

```diff
@@ -1584,15 +1584,16 @@ private static function convertShorthand($val)
     {
         $val = trim($val);
         $last = strtolower($val[strlen($val) - 1]);
-
-        $val = (int) $val;
+        if (!is_numeric($last)) {
+            $val = substr($val, 0, strlen($val) - 1);
+        }
 
         switch ($last) {
             // The 'G' modifier is available since PHP 5.1.0
             case 'g':
-                $val *= 1024;
+                $val *= 1024 * 1024 * 1024;
```

Comment:
Would you mind adding a test for this switch? Without break statements it seems wrong for me here? What do you think?